src/python/pants/core/util_rules/partitions.py

```diff
@@ -36,7 +36,31 @@ class PartitionerType(Enum):
     """The plugin author has a rule to go from `RequestType.PartitionRequest` -> `Partitions`."""
 
     DEFAULT_SINGLE_PARTITION = "default_single_partition"
-    """Registers a partitioner which returns the inputs as a single partition."""
+    """Registers a partitioner which returns the inputs as a single partition
+
+    The returned partition will have no metadata."""
+
+    DEFAULT_ONE_PARTITION_PER_INPUT = "default_one_partition_per_input"
+    """Registers a partitioner which returns a single-element partition per input.
+
+    Each of the returned partitions will have no metadata."""
+
+    def default_rules(self, cls, *, by_file: bool) -> Iterable:
+        if self == PartitionerType.CUSTOM:
+            # No default rules.
+            return
+        elif self == PartitionerType.DEFAULT_SINGLE_PARTITION:
+            if by_file:
+                yield from _single_partition_field_sets_by_file_partitioner_rules(cls)
+            else:
+                yield from _single_partition_field_sets_partitioner_rules(cls)
```

Comment:
I'd prefer:
```python
rules_generator = ... if by_file else ...
yield from rules_generator(cls)
```

we can shorten the names, since now they are truly private to this module :smile: